### ISSUE TITLE
python3.6 dropped but still requires python >= 3.6

### ISSUE THREAD

**Issue by JonathanRenon-EDB** [jmcgeheeiv/pyfakefs#707]
**Describe the bug**
Hello,
support for python 3.6 is dropped since 4.6.0 but requirement still allows for python3.6 to be used with latest versions of pyfakefs.
This allows pip to list and install latest pyfakefs on python3.6.

I did not run any test to verify that latest was effectively still working with python 3.6.

**Comment by JonathanRenon-EDB** [comment]
https://pypi.org/project/pyfakefs/
still shows python>=3.6 
in addition our dependabot running on python3.6 returned latest version (4.6.3) as an update for our current requirements (4.5.6)
extract from the dependabot run:
getting python 3.6.15
```
  proxy | 2022/08/22 10:13:04 [072] GET https://www.python.org:443/ftp/python/3.6.15/Python-3.6.15.tar.xz
```
submitting a PR for pyfakefs 4.6.3
```
updater | INFO <job_443473186> Updating pyfakefs from 4.5.6 to 4.6.3
  proxy | 2022/08/22 10:18:39 [204] GET https://pypi.org:443/simple/pip/
  proxy | 2022/08/22 10:18:39 [204] 200 https://pypi.org:443/simple/pip/
  proxy | 2022/08/22 10:18:40 [206] GET https://pypi.org:443/simple/pip/
  proxy | 2022/08/22 10:18:40 [206] 200 https://pypi.org:443/simple/pip/
  proxy | 2022/08/22 10:18:43 [208] GET https://pypi.org:443/simple/pip/
  proxy | 2022/08/22 10:18:43 [208] 200 https://pypi.org:443/simple/pip/
  proxy | 2022/08/22 10:18:45 [210] GET https://pypi.org:443/simple/setuptools/
  proxy | 2022/08/22 10:18:45 [210] 304 https://pypi.org:443/simple/setuptools/
  proxy | 2022/08/22 10:18:46 [212] GET https://pypi.org:443/pypi/pyfakefs/json
  proxy | 2022/08/22 10:18:46 [212] 200 https://pypi.org:443/pypi/pyfakefs/json
  proxy | 2022/08/22 10:18:48 [214] GET https://pypi.org:443/simple/setuptools/
  proxy | 2022/08/22 10:18:48 [214] 304 https://pypi.org:443/simple/setuptools/
  proxy | 2022/08/22 10:18:49 [216] GET https://api.github.com:443/repos/EnterpriseDB/tpaexec/commits?per_page=100
  proxy | 2022/08/22 10:18:49 [216] * authenticating github api request
  proxy | 2022/08/22 10:18:49 [216] 200 https://api.github.com:443/repos/EnterpriseDB/tpaexec/commits?per_page=100
  proxy | 2022/08/22 10:18:50 [218] GET https://pypi.org:443/pypi/pyfakefs/json
  proxy | 2022/08/22 10:18:50 [218] 200 https://pypi.org:443/pypi/pyfakefs/json
  proxy | 2022/08/22 10:18:50 [220] GET https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/releases?per_page=100
  proxy | 2022/08/22 10:18:50 [220] * authenticating github api request
  proxy | 2022/08/22 10:18:50 [220] 200 https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/releases?per_page=100
  proxy | 2022/08/22 10:18:50 [222] GET https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/contents/
  proxy | 2022/08/22 10:18:50 [222] * authenticating github api request
  proxy | 2022/08/22 10:18:50 [222] 200 https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/contents/
  proxy | 2022/08/22 10:18:51 [224] GET https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/contents/docs
  proxy | 2022/08/22 10:18:51 [224] * authenticating github api request
  proxy | 2022/08/22 10:18:51 [224] 200 https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/contents/docs
  proxy | 2022/08/22 10:18:51 [226] GET https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/contents/CHANGES.md?ref=master
  proxy | 2022/08/22 10:18:51 [226] * authenticating github api request
  proxy | 2022/08/22 10:18:51 [226] 200 https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/contents/CHANGES.md?ref=master
  proxy | 2022/08/22 10:18:51 [228] GET https://github.com:443/jmcgeheeiv/pyfakefs.git/info/refs?service=git-upload-pack
  proxy | 2022/08/22 10:18:51 [228] * authenticating git server request (host: github.com)
  proxy | 2022/08/22 10:18:51 [228] 200 https://github.com:443/jmcgeheeiv/pyfakefs.git/info/refs?service=git-upload-pack
  proxy | 2022/08/22 10:18:51 [230] GET https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.5.6
  proxy | 2022/08/22 10:18:51 [230] * authenticating github api request
  proxy | 2022/08/22 10:18:51 [230] 200 https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.5.6
  proxy | 2022/08/22 10:18:51 [232] GET https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.6.3
  proxy | 2022/08/22 10:18:51 [232] * authenticating github api request
  proxy | 2022/08/22 10:18:52 [232] 200 https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.6.3
  proxy | 2022/08/22 10:18:52 [234] GET https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.5.6
  proxy | 2022/08/22 10:18:52 [234] * authenticating github api request
  proxy | 2022/08/22 10:18:52 [234] 200 https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.5.6
  proxy | 2022/08/22 10:18:52 [236] GET https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.6.3
  proxy | 2022/08/22 10:18:52 [236] * authenticating github api request
  proxy | 2022/08/22 10:18:52 [236] 200 https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.6.3
  proxy | 2022/08/22 10:18:52 [238] GET https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.5.6
  proxy | 2022/08/22 10:18:52 [238] * authenticating github api request
  proxy | 2022/08/22 10:18:52 [238] 200 https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.5.6
  proxy | 2022/08/22 10:18:52 [240] GET https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.6.3
  proxy | 2022/08/22 10:18:52 [240] * authenticating github api request
  proxy | 2022/08/22 10:18:53 [240] 200 https://api.github.com:443/repos/jmcgeheeiv/pyfakefs/commits?sha=v4.6.3
updater | INFO <job_443473186> Submitting pyfakefs pull request for creation
```

**Comment by mrbean-bremen** [comment]
Thanks, interesting - I don't know how this is determined (I thought it would be sufficient to remove Python 3.6 from `setup.py`). I won't have the time to check this today, will probably have a closer look tomorrow.

**Comment by JonathanRenon-EDB** [comment]
Thanks !

**Comment by mrbean-bremen** [comment]
There is indeed still a `python_requires='>=3.6'` in `setup.py` - for some reason I completely overlooked that...
Will fix that in a moment.
The only change we made that is not Python 3.6 compatible was removing the support for `stat_float_times`, which was deprecated since Pyton 3.3 and hardly ever used - I think we can live with that without yanking the versions from PyPi. The main reason we removed it is that the support for Python 3.6 in the CI and other tools is removed, so we cannot reliably test in anymore.
Thanks for the catch!
